src/network/network_server.cpp

```diff
@@ -620,26 +658,12 @@ NetworkRecvStatus ServerNetworkGameSocketHandler::SendMap()
 			 *  to send it is ready (maybe that happens like never ;)) */
 			this->status = STATUS_DONE_MAP;
 
-			/* Find the best candidate for joining, i.e. the first joiner. */
-			NetworkClientSocket *best = nullptr;
+			this->CheckNextClientToSendMap();
+		} else if (std::chrono::steady_clock::now() > last_time + std::chrono::seconds(1)) {
```

Comment:
This only helps if downloading client is actually active. If he just went mia whole queue will can get disconnected before network.max_download_time kicks in for downloading client.